### PR TITLE
Stop combat release kick when player cannot manoverboard

### PR DESCRIPTION
**Pull request by HumorousFool**
This PR adds a check to see whether the craft is disabled, in another world or out of range of the pilot's manoverboard range.

### CODE REVIEW

File: src/main/java/net/countercraft/movecraft/combat/movecraftcombat/status/StatusManager.java [hunk 1]
```diff
@@ -100,31 +101,36 @@ public void craftReleased(@NotNull CraftReleaseEvent e) {
...
-            }.runTaskLater(MovecraftCombat.getInstance(), 5);
-        }
-        if(Config.CombatReleaseBanLength > 0 || Config.EnableCombatReleaseKick) {
-            new BukkitRunnable() {
-                @Override
-                public void run() {
-                    if(player == null || !player.isOnline()) {
-                        return;
+
+        if(manOverboard) {
```
Comment (TylerS1066) [inline src/main/java/net/countercraft/movecraft/combat/movecraftcombat/status/StatusManager.java:113]: To simplify and streamline the code performance, let's change this to the following format:
```java
if(Config.CombatReleaseBanLength <= 0 && !Config.EnableCombatReleaseKick)
  return;
if(!canOverboard())
  return;
if(Config.CombatReleaseBanLength > 0) {
  // do ban
}
// do kick
```

File: src/main/java/net/countercraft/movecraft/combat/movecraftcombat/status/StatusManager.java [hunk 2]
```diff
@@ -175,4 +181,17 @@ private boolean isInAirspace(@NotNull Craft craft) {
...
+    private boolean canManOverboard(Player player, Craft craft) {
+        if(craft.getDisabled())
+            return false;
+        if (craft.getW() != player.getWorld())
+            return false;
+
+        double telX = (craft.getHitBox().getMinX() + craft.getHitBox().getMaxX())/2D;
+        double telZ = (craft.getHitBox().getMinZ() + craft.getHitBox().getMaxZ())/2D;
+        double telY = craft.getHitBox().getMaxY() + 1;
+        Location telPoint = new Location(craft.getW(), telX, telY, telZ);
```
Comment (TylerS1066) [inline src/main/java/net/countercraft/movecraft/combat/movecraftcombat/status/StatusManager.java:194]: To simplify this we could merely use craft.getHitBox.getMidpoint, which is an approximation but should be sufficient and much simpler.

File: src/main/java/net/countercraft/movecraft/combat/movecraftcombat/status/StatusManager.java [hunk 3]
```diff
@@ -175,4 +181,17 @@ private boolean isInAirspace(@NotNull Craft craft) {
...
+        return !(telPoint.distanceSquared(player.getLocation()) > Settings.ManOverboardDistSquared);
```
Comment (TylerS1066) [inline src/main/java/net/countercraft/movecraft/combat/movecraftcombat/status/StatusManager.java:195]: To simplify this, `return telpointdistsquared =< settingsdistsquared`.

**Review by TylerS1066 — CHANGES_REQUESTED** [review]
I like this solution, but in the future I would prefer that we open issues to discuss changes like this before coding them (in the case where they are not wanted).

There are some slight edits I'd like to have made to improve performance, but other than that looks good!

**Comment by HumorousFool** [comment]
Requested changes made, I'll make sure to open issues in the future.

**Review by TylerS1066 — APPROVED** [review]

**Pull request merged**